service/cziti/mfa.go

```diff
@@ -80,12 +81,17 @@ func ziti_mfa_enroll_cb_go(_ C.ziti_context, status C.int, enrollment *C.ziti_mf
 }
 
 // requires that the identity already be fully authenticated and used specifically for enrollment
-func VerifyMFA(id *ZIdentity, code string) {
+func VerifyMFA(id *ZIdentity, code string) error {
 	ccode := C.CString(code)
 	defer C.free(unsafe.Pointer(ccode))
 
 	log.Tracef("verifying MFA for fingerprint: %s using code: %s", id.Fingerprint, code)
 	C.ziti_mfa_verify(id.czctx, ccode, C.ziti_mfa_cb(C.ziti_mfa_cb_verify_go), unsafe.Pointer(C.CString(id.Fingerprint)))
+	authVerifyResult := strings.TrimSpace(<-mfaAuthVerifyResults)
```

Comment:
this changes the function to effectively be blocking. i assume that you've tested all the mfa scenarios? this along with the change on line 117 have me worried. as long as you feel like this logic is 'fine' and tested, then this change is intentional?

Reply:
I have tested it. It looks fine